docs/rfc/rfc-2.adoc

```diff
@@ -0,0 +1,799 @@
+state. This RFC revises that behavior to only archive on transition to CLOSED.
+When contracts transition into TERMINATED state and that transition is
+confirmed (by waiting
+https://github.com/keep-network/keep-ecdsa/blob/master/pkg/client/client.go#L28-L31[`blockConfirmation`
+blocks]), the client should immediately begin <<bitcoin-fee,Bitcoin fee
+resolution>> followed by the <<signing-protocol>> and finally
+<<transaction-submission,transaction submission>>. Only when signing and
+transaction submission (whether via logging or API) completes or times out
+should the associated key material be archived.
+
+On startup, if a client sees that an unarchived keep is in the TERMINATED
+state and has been in that state for `blockConfirmation` blocks, it should
+move directly to <<bitcoin-fee,Bitcoin fee resolution>>, unless an optional
+Bitcoin node connection is available (see <<electrs>>). In this case, the
+client should check the keep's associated deposit for an unspent balance by
+issuing a GET request to `/tx/:txid/outspend/:vout`, using the deposit's
+`fundingInfo().utxoOutpoint` to derive the transaction id and vout, and only
+start the signing process if the JSON result has a `spent` value of false. If
+`spent` is `true` and `status.confirmed` is `false`, a retry should occur
+after an hour until `spent` is `true`, `status.confirmed` is `true`, and
+`status.block_height` is at least 6 blocks past the value returned by issuing
+a GET request to `/blocks/tip/height`. If this occurs, the key material
+should be archived.
+
+Note that the output spend may go from being `spent` and unconfirmed to being
+unspent if the transaction spending the UTXO is dropped from the mempool
+before being included in a block, at which point the bitcoin fee resolution
+and signing process should proceed as if the keep was just terminated.
+
+[[bitcoin-fee]]
+===== Dealing with the Bitcoin Fee
+
+The Bitcoin fee used for clients is something all three signers must agree
+on, since it is part of the sighash. To agree on a fee, this proposal
+suggests adding a single key to the `Extensions.TBTC.BTCRefunds` config
+entry, `MaxFeePerVByte`, which is a value in sats per vByte that represents
+the maximum fee per vByte this client is willing to pay in refund
+transactions. The default for this value should be 75 sats per vByte.
+
+If an optional Bitcoin connection is available (see <<electrs>>), the client
+should look up the 25-block suggested fee per vByte and, if it is available
+and nonzero, should replace the hardcoded max fee with this value. 25 blocks
+means confirmation within ~4 hours. The default of 75 sats per vByte is aimed
+to be slightly higher than a reasonable value for 25 blocks, though of course
+fee market fluctuation can change this at a moment's notice. The 25-block
+suggested fee per vByte should be available at the `/fee-estimates` endpoint
+of the REST API, which returns a JSON object with the key `"25"`
+corresponding to this fee.
```

Comment:
Should there be a maximum fee set, for example, 5% of the deposit? Thinking about a scenario where electrs returns very high value and client blindly accepts it. Even if we assume we can trust electrs, having such maximum could make sense for smaller lot sizes where the value returned can be a significant portion of the deposit.

Reply:
My default thought is that this will ultimately prove unnecessary. This is part of why I'm suggesting we look for 25-block confirmation fees. However, liquidations are more likely when BTC is high, and Bitcoin fees are generally higher when BTC is high, so that's not crazy.

Would we treat this as a `min`, basically? `min(first_available(fee_estimate, configured_fee), 5%*utxo value)` or so?

Reply:
> Would we treat this as a min, basically? min(first_available(fee_estimate, configured_fee), 5%*utxo value) or so?

Yeah, so it's basically: I am fine with any fee as long as it's no more than 5%*utxo. 

A separate question is, what if it's more than 5%*utxo. They should probably re-attempt the protocol after some time.

Reply:
I kind of hate that, tbh. Adding a different retry in that scenario is going to add a boatload of complexity IMO---both to code and to understanding.

Reply:
Actually if we just adhere to the existing retry strategy that seems sufficient, no? Even though fees don't change every 2 minutes, the protocol will abort early in that scenario, which seems fine. We'll hit the fee endpoint more frequently than might be “ideal”, but it shouldn't be an overall problem, I don't think. Thoughts?

Reply:
Yeah, I like this approach. 👍

Reply:
Ok, added this, with one small change: if estimated fee yields >5%*utxo outcome, we fall back on `min(estimated, configured)`. There's a world where the estimated fee is smaller than the configured fee and is still >5%*utxo, and I think in this case it makes sense to continue using the estimated fee.